src/mod_webdav.c

```diff
@@ -1207,11 +1208,14 @@ static int webdav_parse_chunkqueue(server *srv, connection *con, handler_ctx *hc
 #endif
 
 #ifdef USE_LOCKS
-static int webdav_lockdiscovery(connection *con, buffer *locktoken, const char *lockscope, const char *locktype, int depth) {
+static int webdav_lockdiscovery(connection *con, buffer *locktoken, const char *lockscope, const char *locktype, const char *owner, int depth) {
 
 	buffer *b = chunkqueue_append_buffer_open(con->write_queue);
 
-	http_header_response_set(con, HTTP_HEADER_OTHER, CONST_STR_LEN("Lock-Token"), CONST_BUF_LEN(locktoken));
+	/* wrap < > around the Lock-Token header */
+	char locktokenheader[buffer_string_length(locktoken) + 3];
+	snprintf(locktokenheader, sizeof(locktokenheader), "<%s>", locktoken->ptr);
```

Comment:
Please do not use snprintf() for something so trivial.  snprintf() is slow.

Should the token be store in the database with the "<" ... ">"?  If not, then please (carefully) construct the string in the buffer before passing it to set the Lock-Token header value.

Reply:
lighttpd (old version) does not store them with "<...>".